### PR TITLE
feat(VSlider): reverse property for VSlider allows for max on left and min on right

### PR DESCRIPTION
**Pull request by sureshvv**
## Markup:

<details>

```vue
// Paste your FULL Playground.vue here
```
</details>

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and breaking changes).
- [x] My code follows the code style of this project.
- [ ] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)

### CODE REVIEW

File: packages/vuetify/src/components/VRangeSlider/VRangeSlider.ts [hunk 1]
```diff
@@ -146,7 +147,7 @@ export default VSlider.extend({
         ref: 'track'
       }, children)
     },
-    genChildren () {
```
Comment (sh7dm) [inline packages/vuetify/src/components/VRangeSlider/VRangeSlider.ts]: no need for strict types on methods

File: packages/vuetify/src/components/VRangeSlider/VRangeSlider.ts [hunk 2]
```diff
@@ -83,7 +84,7 @@ export default VSlider.extend({
   },
 
   methods: {
-    getTrackStyle (startLength: number, endLength: number, startPadding = 0, endPadding = 0) {
+    getTrackStyle (startLength: number, endLength: number, startPadding = 0, endPadding = 0): Partial<CSSStyleDeclaration> {
```
Comment (sh7dm) [inline packages/vuetify/src/components/VRangeSlider/VRangeSlider.ts]: No need for `: Partial<CSSStyleDeclaration>`

File: packages/vuetify/src/components/VRangeSlider/VRangeSlider.ts (hunk 1 above)
Comment (sureshvv) [reply packages/vuetify/src/components/VRangeSlider/VRangeSlider.ts]: Copied from VSlider.ts. Should I leave it there?

File: packages/vuetify/src/components/VRangeSlider/VRangeSlider.ts (hunk 2 above)
Comment (sureshvv) [reply packages/vuetify/src/components/VRangeSlider/VRangeSlider.ts]: Copied from VSlider.ts. Should I leave it there?

Comment (sh7dm) [reply packages/vuetify/src/components/VRangeSlider/VRangeSlider.ts]: no

File: packages/vuetify/src/components/VRangeSlider/VRangeSlider.ts (hunk 1 above)
Comment (sh7dm) [reply packages/vuetify/src/components/VRangeSlider/VRangeSlider.ts]: no

**Comment by jacekkarczmarczyk** [comment]
Why not just check if max < min instead of adding a new prop?

**Comment by DRoet** [comment]
> Why not just check if max < min instead of adding a new prop?

yeah that would be pretty neat, altho it would kind of defeat the purpose of "max" and "min" as concepts. maybe the props should've been called `start` and `end` 🤷‍♂

**Comment by sureshvv** [comment]
> > Why not just check if max < min instead of adding a new prop?
> 
> yeah that would be pretty neat, altho it would kind of defeat the purpose of "max" and "min" as concepts. maybe the props should've been called `start` and `end`

or left and right would be even better :D

**Comment by sureshvv** [comment]
Screenshot from 2019-03-23 11-35-09

This is to give you an idea of why it may be useful. Or if you know how this can be done without a new property.

**Comment by jacekkarczmarczyk** [comment]
@sureshvv Where would you display `left` value in RTL? On the left (like in LTR) or on the right?

**Comment by sureshvv** [comment]
> @sureshvv Where would you display `left` value in RTL? On the left (like in LTR) or on the right?

The beauty of left and right is they don't change with rtl. Left is left is left.

**Comment by jacekkarczmarczyk** [comment]
And the cons are that 
- you'll need another props for vertical slider (there's no left/right)
- i'm not sure about this, so correct me if i'm wrong, but i guess that the default layout in RTL mode is just the mirror image of LTR, so in multilanguage site if you want to switch from LTR to RTL you'll need to switch left and right values

**Comment by sureshvv** [comment]
> And the cons are that
> 
> * you'll need another props for vertical slider (there's no left/right)
> * i'm not sure about this, so correct me if i'm wrong, but i guess that the default layout in RTL mode is just the mirror image of LTR, so in multilanguage site if you want to switch from LTR to RTL you'll need to switch left and right values

1. yep.. may be top and bottom for vertical sliders

2. yep.. we are punting on rtl and let the individual programmer worry about it - but it may be all right for most users. not really sure about that though :)

File: packages/vuetify/src/components/VRangeSlider/VRangeSlider.ts (hunk 2 above)
Comment (sureshvv) [reply packages/vuetify/src/components/VRangeSlider/VRangeSlider.ts]: fixed

File: packages/vuetify/src/components/VRangeSlider/VRangeSlider.ts (hunk 1 above)
Comment (sureshvv) [reply packages/vuetify/src/components/VRangeSlider/VRangeSlider.ts]: fixed

**Comment by sureshvv** [comment]
@sh7dm Can we merge?

**Comment by DRoet** [comment]
Have we settled on the extra prop in the discussion above?

**Comment by sureshvv** [comment]
> Have we settled on the extra prop in the discussion above?

You can check out the discussion on #6814 which was the previous version of this.

**Comment by johnleider** [comment]
This Pull Request is being closed due to inactivity.

Thank you for your contribution and interest in improving Vuetify!

**Comment by sureshvv** [comment]
This is ready to be merged. Please merge.

**Comment by jacekkarczmarczyk** [comment]
There's still no consensus on the props names

**Comment by sureshvv** [comment]
> There's still no consensus on the props names

Let us vote and settle on one please.

**Comment by nekosaur** [comment]
My vote is for whatever we've already used before. Which I think is `reverse`?

**Comment by sureshvv** [comment]
My vote is also for "reverse".

**Comment by KaelWD** [comment]
I'm ok with reverse. The min/max thing sounds confusing, and left/right doesn't make sense for rtl or vertical.

**Comment by jacekkarczmarczyk** [comment]
The problem with `reverse` is that for text fields it aligns label and input texts, it could be used for placing label on the opposite side also in v-slider and selection controls (there's an issue for this afaik).

Definitely I agree that min/max and left/right are not the best, but there is still start/end option, could be also from-value/to-value or just from/to or maybe event something better

**Comment by jacekkarczmarczyk** [comment]
And another thing - with this playground:
```vue
<template>
  <v-app>
    <v-content>
      <v-container>
        <v-slider min="1" max="10" :tick-labels="[1, 2, 3, 4, 5, 6, 7, 8, 9, 10]" />
        <v-slider min="1" max="10" :tick-labels="[1, 2, 3, 4, 5, 6, 7, 8, 9, 10]" reverse />
        <v-slider min="1" max="10" :tick-labels="[1, 2, 3, 4, 5, 6, 7, 8, 9, 10]" vertical />
        <v-slider min="1" max="10" :tick-labels="[1, 2, 3, 4, 5, 6, 7, 8, 9, 10]" vertical reverse />
      </v-container>
    </v-content>
  </v-app>
</template>
```

i got this:

image


So first of all labels are wrong in reverse mode, second - I'd expect that the initial value will be the left side value, so for normal slider it'd be 5, for reverse - 15, third - keyboard navigation with up/down keys is reversed in reverse mode (`up` moves the slider indicator down)

**Comment by sureshvv** [comment]
@jacekkarczmarczyk... your playground works correctly for me now.

**Comment by sureshvv** [comment]
Screenshot from 2019-06-01 00-22-54

**Comment by jacekkarczmarczyk** [comment]
Great, ty, still there are 2 other issues in my comment

**Comment by sureshvv** [comment]
> Great, ty, still there are 2 other issues in my comment

Fixed 1. Can you check the keyboard input again? Seems quite intuitive for me. Let me know if you would like the keys reversed too.

**Comment by jacekkarczmarczyk** [comment]
For me it's not intuitive for me when you press down arrow and slider goes up, don't know about team's opinion. Also i don't recall using vertical slider with keyboard at all, so won't force anything

There also a new issue:
image

File: .gitignore [hunk 3]
```diff
@@ -1,6 +1,7 @@
 .DS_Store
 node_modules/
 .env
+*~
```
Comment (sh7dm) [inline .gitignore]: What's that

**Comment by sureshvv** [comment]
@jacekkarczmarczyk can you explain new issue?

**Comment by jacekkarczmarczyk** [comment]
colors of the track and ticks are inverted in reverse mode

**Comment by sureshvv** [comment]
> colors of the track and ticks are inverted in reverse mode

No idea how that happened :) will check.

**Comment by jacekkarczmarczyk** [comment]
1. `pageDown`/`pageUp` are not handled properly - in vertical mode page UP should move UP, not sure what should be proper behaviour for horizontal
2. `home`/`end` - imho in all cases (vertical/horizontal, and normal/reverse) `home` should set the initial value (5 for normal, 15 for reversed), `end` - the opposite
2. RTL switches top/bottom in vertical slider

I'm starting to think that allowing `min` to be greater than `max` would be a better solution, also in terms of implementation...

File: packages/vuetify/src/components/VSlider/VSlider.ts [hunk 4]
```diff
@@ -549,7 +553,8 @@ export default mixins<options &
       if ([left, right, down, up].includes(e.keyCode)) {
         this.keyPressed += 1
 
-        const increase = this.$vuetify.rtl ? [left, up] : [right, up]
+        const increase = this.decreasing ? this.vertical ? [right, down] : [left, up]
```
Comment (jacekkarczmarczyk) [inline packages/vuetify/src/components/VSlider/VSlider.ts]: that's a bit unreadable, also i wonder if we should handle left/right for vertical slider?

Comment (sureshvv) [reply packages/vuetify/src/components/VSlider/VSlider.ts]: will try to make it more readable

**Comment by sureshvv** [comment]
> 1. `pageDown`/`pageUp` are not handled properly - in vertical mode page UP should move UP, not sure what should be proper behaviour for horizontal
> 2. `home`/`end` - imho in all cases (vertical/horizontal, and normal/reverse) `home` should set the initial value (5 for normal, 15 for reversed), `end` - the opposite

Will check.

> 3. RTL switches top/bottom in vertical slider
 

How can I test RTL in the Playground?

> I'm starting to think that allowing `min` to be greater than `max` would be a better solution, also in terms of implementation...

I think it will boil down to current implementation anyway at the end.

File: .gitignore (hunk 3 above)
Comment (sureshvv) [reply .gitignore]: Backup files created by "vi" editor. Prett standard in .gitignores where I come from :D

**Comment by jacekkarczmarczyk** [comment]
`<v-checkbox v-model=$vuetify.rtl />`

**Comment by nekosaur** [comment]
ping @sureshvv

**Comment by johnleider** [comment]
This issue is being closed due to:

- [ ] Reproduction environment not provided (inactivity for 14 days)
- [x] Requested information not provided (inactivity for 14 days)
- [ ] Violation of the [Code of Conduct](https://github.com/vuetifyjs/vuetify/blob/dev/CODE_OF_CONDUCT.md)

**Pull request closed without merging**